probe_new.py

```diff
@@ -0,0 +1,234 @@
+import asyncio
+import time
+from flask import Flask
+# import helper
+from prometheus_client import Gauge, CollectorRegistry, generate_latest
+import json
+import pykafka
+import logging
+import os
+
+LOGGER = logging.getLogger(__name__)
+
+# the flask app
+app = Flask(__name__)
+
+# create the Gauge object to track the offset change
+registry = CollectorRegistry()
+producerOffsetTracker = Gauge('kafka_offset', 'the latest available offsets', ['service', 'topic', 'partition'],
+                              registry=registry)
+consumerGroupOffsetTracker = Gauge('cg_kafka_offset', 'the consumer group offsets',
+                                   ['service', 'topic', 'partition', 'consumergroup'], registry=registry)
+
+offset_info = {}
+kafka_hosts = "c1-26-usab1-kafka-1.oracle.callstats.io:9092"
+client = pykafka.KafkaClient(hosts=kafka_hosts)
+topic_names = client.topics.keys()
+cgOffsetsInfo_list = []
+
+
+async def say_after(delay, what):
+    await asyncio.sleep(delay)
+    print("finished " + what)
+    print(f"current {time.strftime('%X')}")
+    print(what)
+
+
+@app.route('/internal/test')
+def call_run_main():
+    asyncio.get_event_loop().run_until_complete(run_main())
+    return "OK"
+
+
+async def run_main():
+    tasks = []
+    tasks.append(asyncio.create_task(
+        say_after(1, 'hello')))
+
+    tasks.append(asyncio.create_task(
+        say_after(2, 'world')))
+
+    tasks.append(asyncio.create_task(
+        say_after(5, 'slow')))
+
+    print(f"started at {time.strftime('%X')}")
+
+    # Wait until both tasks are completed (should take
+    # around 2 seconds.)
+
+    # for task in tasks:
+    #    await task
+    # await task1
+    # await task2
+    # await task3
+
+    print(f"finished at {time.strftime('%X')}")
+
+
+@app.route('/internal/metrics')
+def getKafkaOffset():
+    # fetch the latest available offsets from kafka and return the metrics
+    kafka_hosts = "c1-26-usab1-kafka-1.oracle.callstats.io:9092"
```

Comment:
Hardcoded

Reply:
probe_new is not required anymore. Will delete the file. It was kept for testing purpose.